packages/account/src/BiconomySmartAccount.ts

```diff
@@ -74,38 +75,89 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     try {
       this.isProviderDefined()
       this.isSignerDefined()
-      await this.setAccountIndex(accountIndex)
+      this.owner = await this.signer.getAddress()
       this.chainId = await this.provider.getNetwork().then((net) => net.chainId)
+      await this.setAccountIndex(accountIndex)
+      this.isInited = true
     } catch (error) {
-      console.error(`Failed to call init: ${error}`);
+      Logger.error(`Failed to call init: ${error}`);
       throw error
     }
 
     return this
   }
 
+  private isInitialized(): boolean{
+    if (!this.isInited)
+    throw new Error('BiconomySmartAccount is not initialized. Please call BiconomySmartAccount.init() before interacting with any other function')
```

Comment:
BiconomySmartAccount.init() in message seems like init() is static function in the class.

We can modify the message like this

..... Please call init() on BiconomySmartAccount instance before interacting ......

Reply:
resolved